fill_templates.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+#FUNTION DECLARATION ###############################################################################
+createmonconfig(){
+  #function to read all processes from the processes.csv 
+  #and build the array
+  echo -n "Creating output..."
+  cat $1|while read line;do
+    conf=($line)
+    procs=$(eval "echo \"${conf[0]}\"")
+    startstopsc=$(eval "echo \"${conf[1]}\"")
+    output="$configs/${conf[2]}"
+
+    proclist=`tail -n +2 ${procs} | awk -F "\"*,\"*" '{print $3 " " $4}'|cut -d" " -f1,2`
+    echo "$proclist"|while read procs;do 
+      array=($procs)
+      proctype=${array[0]}
+      procname=${array[1]}
+      eval "echo $2" >> $output
+      echo "" >> $output
+    done
+  done 
+  echo " DONE"
+}
+
+createmonalert(){
+  if [ -f ${configs}monitalert.cfg ];then 
+    echo -n "Deleting monitalert.cfg..."
+    rm ${configs}monitalert.cfg
+    echo " DONE" 
+  fi 
+  
+  echo -n "Copying monitalert from ${templates}..."
+  cp ${templates}monitalert.cfg ${configs}
+  echo " DONE"
+}
+
+#SETTING DEFAULT ENVIRONMENT VARIABLES #############################################################
+if [ "-bash" = $0 ]; then
+    dirpath="${BASH_SOURCE[0]}"
+else
+    dirpath="$0"
+fi
+                            
+eval ". $(dirname "$dirpath")/setenv.sh"                                                            #set environment variables  
```

Comment:
why is `eval` needed here? Why not just source the file with `.` directly? (this pattern exists in a few places)

Reply:
This was already here. I think eval is used to execute `". $(dirname "$dirpath")/setenv.sh"`

Reply:
Not sure what you mean by "already here", this is a new file, no? I get what `eval` is doing, I'm asking _why_ though? Seems like you could just do `. $(dirname "$dirpath")/setenv.sh`